### ISSUE TITLE
[a11y]: DataTable [dynamic] - keyboard nav bug 

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#14263]
### Package

@carbon/react

### Browser

FF

### Package version

@carbon/react@1.32.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS

### Description
Unable to move to the previous / next rows using the table navigation command with JAWS

When we use the table navigation commands i.e. control + alt  +Up / down Arrow keys, We are unable to move to the previous / next row. 

Note: It works fine when we are in the first column. The issue happens from the second column onwards. 
Note: This issue is observed in "expansion" and Batch Expansion variants as well.

### Recommended Fix
User should be able to navigate through the table Rows using the table navigation commands. 

When the standard keyboard navigation method is not supported, it impacts the user’s ability to interact with page controls. This affects screen reader users. 

### WCAG 2.1 Violation

2.1.1 Keyboard navigation

### Reproduction/example
Record `Number 67` in the below table:
https://airtable.com/shrFsaeHhImOkuVNR

https://react.carbondesignsystem.com/?path=/story/components-datatable-dynamic--default

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
